File: server/src/main/java/com/example/demo/Entity/ParkingSpace.java
package com.example.demo.Entity;


import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;

@Entity
@Table(name = "ParkingSpace")
public class ParkingSpace {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    private Long id;

    @NotEmpty
    @Column(name = "position")
    private String position;

    @Column(name = "isFree")
    private boolean isFree;

    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.DETACH, CascadeType.REFRESH})
    @JoinColumn(name = "place_id")
    @JsonIgnore
    private Place place;

    public ParkingSpace() {
    }

    public ParkingSpace(@NotEmpty String position, boolean isFree) {
        this.position = position;
        this.isFree = isFree;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getPosition() {
        return position;
    }

    public void setPosition(String position) {
        this.position = position;
    }

    public boolean isFree() {
        return isFree;
    }

    public void setFree(boolean free) {
        isFree = free;
    }

    public Place getPlace() {
        return place;
    }

    public void setPlace(Place place) {
        this.place = place;
    }
}

File: client/lib/providers/parking_space.dart
class ParkingSpace {
  int id;
  String position;
  bool isFree;

  ParkingSpace({
    this.id,
    this.position,
    this.isFree,
  });

  ParkingSpace.fromJson(Map<String, dynamic> json) {
    this.id = json['id'];
    this.position = json['position'];
    this.isFree = json['free'];
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'position': position,
        'free': isFree,
      };
}
